### PR TITLE
Deprecated ONCE_INIT in favor of Once::new()

### PR DESCRIPTION
**Pull request by lzutao**
cc rust-lang/rust#61757

### CODE REVIEW

File: tests/testsuite/support/paths.rs [hunk 1]
```diff
@@ -51,18 +51,16 @@ thread_local! {
 }
 
 pub struct TestIdGuard {
-    _private: ()
+    _private: (),
```
Comment (ehuss) [inline tests/testsuite/support/paths.rs]: I would prefer if this were done in a separate PR.

**Comment by lzutao** [comment]
Done.

**Comment by ehuss** [comment]
Thanks!

@bors r+
